### PR TITLE
Fix filters in ExceptionRaceConditionIT

### PR DESCRIPTION
**Pull request by cocreature**
Unfortunately I messed up all the filters (probably got confused by
the very generic sounding names) and it resulted in us just asserting
on nothing.

I am a bit worried about the fragility of those tests. If you mistype
a choice name you run into the same issue. A potentially more robust
option might be to assert that the filters are non-empty or run them
with the assertion reversed and check that they fail but that seems
out of scope for this PR.

I did reverse all assertions manually and checked that the tests do
fail so they properly assert on what they should.

Very sorry for messing this up

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ExceptionRaceConditionIT.scala [hunk 1]
```diff
@@ -173,3 +175,93 @@ final class ExceptionRaceConditionIT extends LedgerTestSuite {
...
+
+    private def isCreated(templateName: String)(event: TreeEvent): Boolean =
+      event.kind.isCreated && event.getCreated.templateId.exists(_.entityName == templateName)
+
+    private def isExerciseEvent(choiceName: String)(event: TreeEvent): Boolean =
+      event.kind.isExercised && event.getExercised.choice == choiceName
+
+    // def isCreateDummyContract(tx: TransactionTree): Boolean =
+    //   tx.containsEvent(isCreated(RaceTests.DummyContract.TemplateName))
+
```
Comment (fabiotudone-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ExceptionRaceConditionIT.scala:198]: Same here

**Review by fabiotudone-da — COMMENTED** [review]
Thanks! Even though the utilities are now test-scoped, I'd consider either using more specific method names or having few general ones and passing a simple, specific predicate to them directly from the assertions.

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ExceptionRaceConditionIT.scala [hunk 2]
```diff
@@ -173,3 +175,93 @@ final class ExceptionRaceConditionIT extends LedgerTestSuite {
...
+  private object ExceptionRaceTests {
+    object ContractWithKey {
+      val TemplateName = "ContractWithKey"
+      val ChoiceArchive = "ContractWithKey_Archive"
+//      val ChoiceExercise = "ContractWithKey_Exercise"
+    }
+
+    // object DummyContract {
+    //   val TemplateName = "DummyContract"
+    // }
```
[the remark this answers is not in the document]
Comment (cocreature) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ExceptionRaceConditionIT.scala]: :facepalm: 2391682ea869405c9837b904e4c5b2e151305c8a

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ExceptionRaceConditionIT.scala (hunk 1 above)
Comment (cocreature) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ExceptionRaceConditionIT.scala]: 2391682ea869405c9837b904e4c5b2e151305c8a

**Comment by cocreature** [comment]
> Thanks! Even though the utilities are now test-scoped, I'd consider either using more specific method names or having few general ones and passing a simple, specific predicate to them directly from the assertions.

Is 3dbb8b0817cf8d14bfe0ceffc18eec9519539927 along the lines of what you were suggesting?

I’m not sure I understand the non exception tests well enough to do the same for those (never worked on them or looked at them in detail) so I’d prefer to leave that out of this PR.

**Comment by fabiotudone-da** [comment]
> Is [3dbb8b0](https://github.com/digital-asset/daml/commit/3dbb8b0817cf8d14bfe0ceffc18eec9519539927) along the lines of what you were suggesting?

Yes, thanks; I think they could also be factored back now but that's a nit.

**Review by fabiotudone-da — APPROVED** [review]

**Comment by cocreature** [comment]
> Yes, thanks; I think they could also be factored back now but that's a nit.

good point, probably makes sense to do this once we’ve switched over the other tests as well. I’ll see if I can figure those tests out well enough to do that but probably not before next week and happy if someone beats me to it.

**Pull request merged**